### PR TITLE
PyHSPlasma SDL

### PR DESCRIPTION
**Pull request by Hoikas**
This adds just enough SDL to PyHSPlasma for one to be able to poke around the SDL files.

### CODE REVIEW

File: Python/SDL/pyStateDescriptor.cpp [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        return 0;
+    }
+
+    PyErr_SetString(PyExc_TypeError, "StateDescriptor subscript should be an int or a string");
+    return -1;
+}
+
+PY_PLASMA_REPR_DECL(StateDescriptor)
+{
+    return PyString_FromSTString(self->fThis->getName());
```
Comment (zrax) [inline Python/SDL/pyStateDescriptor.cpp]: The `repr` string should include more than just the name of the descriptor...  Maybe something like `<plStateDescriptor {name}>`?

File: Python/SDL/pyStateDescriptor.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+PY_METHOD_VA(StateDescriptor, delVariable,
+             "Params: variable\n"
+             "Deletes the requested variable by name or index")
+{
+    Py_ssize_t idx;
+    if (PyArg_ParseTuple(args, "n", &idx)) {
+        try {
+            self->fThis->delVariable(idx);
+        } catch (...) {
```
Comment (zrax) [inline Python/SDL/pyStateDescriptor.cpp]: Same comments as above

File: Python/SDL/pyVarDescriptor.cpp [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+#include <SDL/plStateDescriptor.h>
+
+PY_PLASMA_DEALLOC(VarDescriptor)
+PY_PLASMA_EMPTY_INIT(VarDescriptor)
+PY_PLASMA_NEW(VarDescriptor, plVarDescriptor)
+
+PY_PLASMA_REPR_DECL(VarDescriptor)
+{
+    return PyString_FromSTString(self->fThis->getName());
```
Comment (zrax) [inline Python/SDL/pyVarDescriptor.cpp]: Similar to above, maybe `<plVarDescriptor {name}>`?

File: Python/SDL/pyStateDescriptor.cpp (hunk 1 above)
Comment (Hoikas) [reply Python/SDL/pyStateDescriptor.cpp]: This is actually the implementation of the `__str__()` method, which shares its signature with `__repr__()`. I can introduce a proper `__repr__()` if you prefer.

Comment (zrax) [reply Python/SDL/pyStateDescriptor.cpp]: Sneaky; I didn't catch that you were assigning it to `tp_str` :trollface:.  In that case, I might prefer a new macro (even if it's almost identical to `PY_PLASMA_REPR_DECL`) to make it clearer in the code.

**Comment by Hoikas** [comment]
Updated to fix issues from review.

**Pull request merged**